src/pages/home/report/ReportActionItem.js

```diff
@@ -70,59 +69,47 @@ const propTypes = {
 
 const defaultProps = {
     draftMessage: '',
-    hasOutstandingIOU: false,
 };
 
-class ReportActionItem extends Component {
-    constructor(props) {
-        super(props);
-        this.popoverAnchor = undefined;
-        this.state = {
-            isContextMenuActive: ReportActionContextMenu.isActiveReportAction(props.action.reportActionID),
-        };
-        this.checkIfContextMenuActive = this.checkIfContextMenuActive.bind(this);
-        this.showPopover = this.showPopover.bind(this);
-        this.renderItemContent = this.renderItemContent.bind(this);
-        this.toggleReaction = this.toggleReaction.bind(this);
-    }
+function ReportActionItem(props) {
+    const [isContextMenuActive, setIsContextMenuActive] = useState(ReportActionContextMenu.isActiveReportAction(props.action.reportActionID));
+    const previousDraftMessageRef = useRef('');
+    const textInputRef = useRef();
+    const popoverAnchorRef = useRef();
 
-    shouldComponentUpdate(nextProps, nextState) {
-        return this.props.displayAsGroup !== nextProps.displayAsGroup
-            || this.props.draftMessage !== nextProps.draftMessage
-            || this.props.isMostRecentIOUReportAction !== nextProps.isMostRecentIOUReportAction
-            || this.props.hasOutstandingIOU !== nextProps.hasOutstandingIOU
-            || this.props.shouldDisplayNewMarker !== nextProps.shouldDisplayNewMarker
-            || !_.isEqual(this.props.action, nextProps.action)
-            || this.state.isContextMenuActive !== nextState.isContextMenuActive;
-    }
+    // We need to store the previous value of draftMessage because we use it in the subsequent useEffect hook
+    // to decide whether we should focus the text input
+    useEffect(() => {
+        previousDraftMessageRef.current = props.draftMessage;
+    }, [props.draftMessage]);
```

Comment:
what's the order of operations and should it be after the next `useEffect()`? If we do this here wouldn't the `previousDraftMessageRef.current` always equal `props.draftMessage` when the other `useEffect()` call runs?

Reply:
I'm not too sure, at least from my testing it doesn't see like the behavior was affected in any way.

But you do make a good point, if this callback is registered first and runs before the other useEffect callback then we'll have set it to the current value before using it.

Reply:
updated